### ISSUE TITLE
ignore a module during build / runtimeModules Option

### ISSUE THREAD

**Issue by jd1378** [nuxt/nuxt.js#7033]
### What problem does this feature solve?
**TL;DR**
there's no way to ignore a module during build command.

Problem:
Currently I have a module which setups a redis client that subscribes to a channel for changes, do a few requests to load the data needed for my index page and redo the requests only if index data has changed, which I only want to run once during runtime for SSR for improving the performance by not doing any network request on SSR, I'm currently passing data from module to my index page SSR requests  using mutating the Vue prototype (I can't think of a proper way to do so right now), but currently I can't find a way to tell Nuxt to ignore the module during build command, and thus if the redis server and ... is not available during build command , it will fail. (plus the build will continue running for a little bit because of this module)

### What does the proposed changes look like?

Add an option like `runtimeModules`/`SSRModules`/etc in addition to `modules` and `buildModules` ? I don't think it be a breaking change overall

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10338">#c10338</a>)</em></sub></div>

**Comment by jd1378** [comment]
For those who may come here in future:

you can do this easily already using hooks inside your module:
`this.nuxt.hook('listen', async () => {});`
